### PR TITLE
Added Support for reading and writing UTF-8 arrays

### PR DESCRIPTION
**Pull request by mstier**
Added support for reading and writing UTF-8 arrays from C++ client. Aimed to enhance interoperability with Java clients.

### CODE REVIEW

**Comment by ihsandemir** [comment]
Hello mstier,

Can you rebase and resubmit the PR? Can you also add a test for this so that we can merge it? Thanks for the PR.

**Comment by sancar** [comment]
verify

**Comment by sancar** [comment]
Hi @mstier , 
To be able to accept your pr we need you to sign our contributor agreement. You can find it here.
https://hazelcast.atlassian.net/wiki/spaces/COM/pages/6357071/Hazelcast+Contributor+Agreement

File: hazelcast/include/hazelcast/client/spi/DefaultObjectNamespace.h [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #define HAZELCAST_CLIENT_SPI_DEFAULTOBJECTNAMESPACE_H_
 
 #include <string>
+#include <functional>
```
Comment (ihsandemir) [inline hazelcast/include/hazelcast/client/spi/DefaultObjectNamespace.h]: why?

File: hazelcast/src/hazelcast/client/serialization/pimpl/PortableReaderBase.cpp [hunk 2]
```diff
@@ -138,6 +138,12 @@ namespace hazelcast {
                     return dataInput.readShortArray();
                 }
 
+
+        				std::auto_ptr<std::vector<std::string> > PortableReaderBase::readUTFArray(const char *fieldName) {
```
Comment (ihsandemir) [inline hazelcast/src/hazelcast/client/serialization/pimpl/PortableReaderBase.cpp]: line format

File: hazelcast/include/hazelcast/client/spi/DefaultObjectNamespace.h (hunk 1 above)
Comment (mstier) [reply hazelcast/include/hazelcast/client/spi/DefaultObjectNamespace.h]: As far as I remember, there was a compile error using the Visual Studio 2015 compiler under Windows without this include. Not sure if this is required anymore.

**Comment by ihsandemir** [comment]
@mstier Are you still working on this PR? We need to add a test to verify the changes and also we need to rebase the PR, can you do them?

**Comment by ihsandemir** [comment]
Fix is merged by https://github.com/hazelcast/hazelcast-cpp-client/pull/596

**Pull request closed without merging**